### ISSUE TITLE
DOC: Measurement modeling - construct reliability

### ISSUE THREAD

**Issue by mmadaio** [fairlearn/fairlearn#799]
#### Describe the issue linked to the documentation
The user guide currently does not include explanations of key concepts from measurement modeling.

Measurement modeling is a framework from quantitative social sciences, involving a process of measuring theoretical constructs that cannot be observed directly (such as fairness) by making inferences from observable properties.

Because fairness itself is unobservable (and is fundamentally contested), users of Fairlearn need to make decisions about what their theoretical understanding of fairness is, and how they operationalize it via observable properties. At each stage, assumptions are made, which need to be tested.

Measurement modeling provides a language and a framework for making those assumptions explicit and testing them.

As many users of Fairlearn may not have quantitative social science backgrounds, we need to provide explanations and examples of the process of measurement modeling, including key concepts such as construct validity and construct reliability. 

This issue is focused on construct reliability (to keep the issues smaller and more achievable). Issue #769 focuses on construct validity.

See Jacobs and Wallach (2021) for more information.

#### Suggest a potential alternative/fix
Create a glossary explaining key terms from construct reliability, with explanations of what they might mean for measuring fairness.

- Test-retest reliability
- Internal consistency reliability

**Comment by alliesaizan** [comment]
@mmadaio do you recommend any particular sources/academic papers to use for definitions of these concepts?

**Comment by mmadaio** [comment]
Yes! So, [Jacobs and Wallach (2021)](https://dl.acm.org/doi/pdf/10.1145/3442188.3445901), section 3.1 is a good place to start!

These are some other good resources:
- https://www.researchgate.net/profile/Veronica-Benet/publication/232498571_MeasurementReliability_construct_validation_and_scale_construction/links/02bfe512de8d72d795000000/MeasurementReliability-construct-validation-and-scale-construction.pdf
- https://onlinelibrary.wiley.com/doi/abs/10.1002/9781119111931.ch15
